imapclient/imapclient.py

```diff
@@ -1206,6 +1207,38 @@ def fetch(self, messages, data, modifiers=None):
         typ, data = self._imap._untagged_response(typ, data, 'FETCH')
         return parse_fetch_response(data, self.normalise_times, self.use_uid)
 
+    def fetch_email_messages(self, messages, _class=None):
+        """Retrieve messages as Python email Messages.
+
+        *messages* is a list of message IDs to retrieve.
+
+        The optional *_class* is passed to the `email.parser.Parser` that
+        generates the Message.
+
+        A dictionary is returned, indexed by message number. Values are
+        instances of `email.message.Message` or `email.message.EmailMessage`
+        depending on the *_class* used and the version of Python.
```

Comment:
How about:
A dictionary is returned, indexed by message id. By default, the values are instances of `email.message.Message` or `email.message.EmailMessage` depending on the version of Python. If *_class* was specified then values will be instances of that class.